src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy

```diff
@@ -86,7 +87,18 @@ class GatlingPlugin implements Plugin<Project> {
                     gatlingImplementation project.sourceSets.test.output
                 }
             }
+
+            if (project.repositories.isEmpty() || isGatlingUnresolved(project)) {
+                project.repositories {
+                    mavenCentral()
+                }
+            }
         }
     }
+
+    private static boolean isGatlingUnresolved(Project project) {
+        def cfg = project.configurations.gatling.copyRecursive()
+        cfg.incoming.resolutionResult.allDependencies.any { it instanceof UnresolvedDependencyResult }
```

Comment:
Shouldn't a method named `isGatlingUnresolved` check that the UnresolvedDependencyResult's groupId is `io.gatling`?

Reply:
Let me explain. The entire configuration `gatling` contains a single dependency - `io.gatling.highcharts:gatling-charts-highcharts`. So the result of `cfg.incoming.resolutionResult.allDependencies` will containt the galting charts and all transitive dependency required to run it.

In case if gatling itself is resolved, but the transitive dependencies are not - the plugin will not work, because of missing dependencies. 
So to make sure plugin could work without forcing user to add repositories manually - the code checks that gatling and all transitive dependencies are resolves succesfully.

Reply:
I get the intent, but what the name of the method and what it actually does don't match.
It verifies if **any** dependency is unresolved, not a Gatling one or a transitive dependency.
For example, the user could add an extra dependency which would be the one unresolved, or make a typo.
There's probably no way to do better, so I would be in favor of renaming the method `isAnyDependencyUnresolved`.

Also, would it be worth checking if mavenCentral is already there? Or is it a Set?

Reply:
> There's probably no way to do better, so I would be in favor of renaming the method isAnyDependencyUnresolved.

Agree, need to be changed. But let me propose another name - `isGatlingConfigurationResolved()` ?

Reply:
> Also, would it be worth checking if mavenCentral is already there? Or is it a Set?

Don't know actually. Will investigate.

Reply:
> But let me propose another name - isGatlingConfigurationResolved() ?

Really, this method does nothing related to Gatling jars or deps. IMHO, for the sake of maintenance, a method name should reflect what it actually does, not what we would like it to do if we could think of a better solution.

But well, it's not worth arguing over this, pick the name of your liking.

Reply:
`gatling` is the name of [Gradle configuration](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.Configuration.html) created by plugin, that contains gatling chart dependency. I.e. gatling dependencies are declared via such config.

```
dependencies {
   gatling "io.gatling.highcharts:gatling-charts-highcharts:3.3.0"
}
```

See [corredponding code](https://github.com/gatling/gatling-gradle-plugin/blob/master/src/main/groovy/io/gatling/gradle/GatlingPlugin.groovy#L80) of plugin.

So, the name `isGatlingConfigurationResolved` reflects that all dependencies declared in `gatling` configurations are resolved.

Does this makes naming more clear ?

Reply:
OK, let's go with that.

Reply:
Thank you.